examples: Add automated Cross-Framework testing for Web Component (React, Vue, Svelte)

lerrybe opened the pull request:
## Related Issue 
https://github.com/thorvg/thorvg.web/issues/140

## Summary  
Add a simple bash script to test framework compatibility across React, Vue, and Svelte examples.

## What the Script Does

The `scripts/build-test.sh` script:

### **Build Validation**  
- Build the WebAssembly and reference the local thorvg.wasm file as the wasmUrl
- Runs `npm run build` for each framework
- Tracks success/failure for each build

## Usage

```bash
# From root directory location
$ ./build-test.sh
# or 
$ npm run test:build
```

## Screenshots

* Check local WASM reference

| Vue | Svelte | React | 
|:--:|:--:|:--:|
|  <img width="1000" height="1067" alt="image" src="https://github.com/user-attachments/assets/97ed50eb-6e92-4190-8d74-c34e86032d4b" /> |  <img width="1000" height="1067" alt="스크린샷 2025-10-19 오후 5 02 25" src="https://github.com/user-attachments/assets/c62cf5bf-65d1-4b60-9304-e972f9d105bb" /> |  <img width="1000" height="1027" alt="image" src="https://github.com/user-attachments/assets/408bbf71-4c33-4b92-93f8-7702e307d578" /> |  

* Check success/failure cases

| All successful cases | Some failure cases | 
|:--:|:--:|
| <img width="2000" height="153" alt="스크린샷 2025-10-19 오후 5 05 51" src="https://github.com/user-attachments/assets/0835d161-cf0e-43c6-90d9-7542c7549e29" /> | <img width="1000" height="171" alt="스크린샷 2025-10-19 오후 5 08 40" src="https://github.com/user-attachments/assets/79fc099e-beac-4319-b441-e2bfa23da521" /> <img width="1000" height="135" alt="스크린샷 2025-10-19 오후 5 11 58" src="https://github.com/user-attachments/assets/824bf180-4c3e-4e07-ad8f-9ed06a4f22fb" /> |

lerrybe commented on scripts/cross-framework-test/README.md:
````diff
@@ -0,0 +1,87 @@
+All paths are relative to the **project root directory**:
+
+```
+scripts/cross-framework-test/
+├── index.js              # Main entry point
+├── constant/
+│   └── config.js             # Config and framework settings
+├── lib/
+│   ├── build-tester.js       # Main build testing logic
+│   ├── cli.js                # Command line interface
+│   ├── file-system.js        # File system utilities
+│   ├── logger.js             # Logging utility
+│   └── reporter.js           # Results formatting and output
+└── README.md             # This file
````
@tinyjin 
Hello, I’ve drafted a proposal for issue [140](https://github.com/thorvg/thorvg.web/issues/140). The diff is a bit large, but I believe that’s mainly because configuration files are included. If further clarification is needed, I’ll be glad to add comments. Please review with a focus on (1) the example setup and (2) the script structure. Thank you.

tinyjin commented on example/react/package.json:
```diff
@@ -0,0 +1,23 @@
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@thorvg/lottie-player": "^1.0.0-beta.1",
```
wrong version, `v1.0.0-beta.1` is unavailable.

Also invalid approach, in the testing, we expect example run build test with local build.

Please check this with another examples as well.

lerrybe commented:
@tinyjin 

I’ve updated the related parts and pushed the changes.
Also reported an [additional issue](https://github.com/thorvg/thorvg.web/issues/157) I found during the process.
I’d appreciate it if you could review them when possible. Thank you

> **Changes**
>
> 1. Updated shell script to include `npm run build` execution (wasm build + web component build) 
> 2. Adjusted `wasmUrl` in examples to reference the local `dist/thorvg.wasm`
> 3. Updated module dependencies React, Vue, and Svelte examples (last updated: 2025-10-19)

tinyjin commented on example/vue/package.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "vue",
```
ThorVG Vue Example

tinyjin commented on example/react/package.json:
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "react",
```
ThorVG React Example

tinyjin commented on example/svelte/package.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "svelte",
```
ThorVG Svelte Example

tinyjin commented on README.md:
````diff
@@ -375,6 +375,18 @@ Please check these examples in various environments.
+Test framework compatibility by running builds across different frontend frameworks:
+
+```bash
+$ ./scripts/cross-framework-test.sh
````
npm run test:build

related: https://github.com/thorvg/thorvg.web/pull/145/files#r2443085563

tinyjin commented on example/react/package-lock.json:
```diff
@@ -0,0 +1,1806 @@
+{
+  "name": "ThorVG React Example",
```
We use `yarn` as the default package manager in this project. Let’s keep everything consistent.

tinyjin reviewed (commented):
Thank you. It's around corner
